torch/_decomp/decompositions.py

```diff
@@ -3111,7 +3111,14 @@ def compute_source_index(coords: Tensor, size: int) -> Tensor:
         return compute_coordinates(coords_un, size)
 
     N, C, iH, iW = a.shape
-    _, oH, oW, _ = grid.shape
+    _, oH, oW, two = grid.shape
+
+    # Let's expand grid to [N, C, oH, oW, 2]
+    # This allows to generate a single triton cuda kernel instead of two kernels.
+    # Two kernels are due source indices, weights have shape (N, 1, oH, oW), xnumel=N*oH*oW
+    # and output has shape (N, C, oH, oW), xnumel=N*C*oH*oW
+    # Expanding grid to (N, C, oH, oW, two) unifies xnumel to N*C*oH*oW
+    grid = grid.view(N, 1, oH, oW, two).expand(N, C, oH, oW, two)
```

Comment:
If you want, what about
```suggestion
    _, oH, oW, two = grid.shape
    assert two == 2

    # Let's expand grid to [N, C, oH, oW, 2]
    # This allows to generate a single triton cuda kernel instead of two kernels.
    # Two kernels are due source indices, weights have shape (N, 1, oH, oW), xnumel=N*oH*oW
    # and output has shape (N, C, oH, oW), xnumel=N*C*oH*oW
    # Expanding grid to (N, C, oH, oW, 2) unifies xnumel to N*C*oH*oW
    grid = grid.view(N, 1, oH, oW, 2).expand(N, C, oH, oW, 2)
```
for clarity